c10/util/complex.h

```diff
@@ -252,15 +252,14 @@ struct alignas(sizeof(T) * 2) complex {
     T ai = imag_;
     U br = rhs.real();
     U bi = rhs.imag();
-    // not using std::abs because it is not a constexpr and
-    // can cause an error in windows
 
-#ifdef _MSC_VER
-    auto abs_br = br < 0 ? -br : br;
-    auto abs_bi = bi < 0 ? -bi : bi;
-#else
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
isn't `#if defined(__GNUC__)` enough?

Reply:
It says [here](https://stackoverflow.com/questions/28166565/detect-gcc-as-opposed-to-msvc-clang-with-macro) that clang uses `__GNUC__` too.

Reply:
Fair play